steps/25_pretrain/step.md

```diff
@@ -0,0 +1,244 @@
+# API for pre-training and fine-tuning
+
+Contributors: @fkiraly
+
+## High-level summary 
+
+### The Aim
+
+`sktime` now has a number of estimators that can do pre-training, fine-tuning, global learning, and cross-learning.
+
+However, the current API design for these use cases has a few problems, and repeatedly issues have been opened for a rework.
+
+This STEP is about finalizing a good interface for:
+
+* global forecasting
+* pre-training
+* fine-tuning of foundation models
+* zero-shot use of foundation models
+
+References:
+
+* conceptual design issue: https://github.com/sktime/sktime/issues/6580
+* umbrella issue foundation models: https://github.com/sktime/sktime/issues/6177
+* newer issue: https://github.com/sktime/sktime/issues/7838
+
+### requirements
+
+* design covers the above use cases with a simple interface
+* composability - use of sensible pipelines, tuning, etc should be simple and not require major surgery in current compositors
+* downwards and upwards compatibility - design should not impact current extension contracts
+* maintainability: maintaining the framework and estimators with the above capabilities should be simple
+
+### The proposed solution
+
+Our proposed solution adds a new state, and a simple switch.
+
+No new public methods are added beyond this, and signatures of methods are not modified.
+
+Estimators get a third state, "pretraining phase", besides unfitted and fitted.
+
+The solution is best illustrated in the basic vignette below.
+
+### Discussion of current solutions
+
+There are multiple current solutions, all have problems:
+
+#### Global forecasting
+
+Forecasters inheriting from `_BaseGlobalForecaster`.
+
+A `y` is added in the `predict` methods. If this is passed, the `fit` is interpreted
+as a pretraining pass.
+
+Problems:
+
+* some models need to know at `fit` time whether the data is for pretraining.
+Examples: global reduction approaches. Broadcasting.
+* as a general design principle, all `predict` methods would need to be changed to
```

Comment:
I do not understand this. Why do `predict` methods need to get the `fit` arguments?

Reply:
This is subjecte to the current approach, of adding the local batch to `predict`-like methods

In forecasting, for instance, we add the `y` to each and every `predict`-like method.

Also, consider the case of classification, where training happens on pairs of instances, one instanece in `X` pairing with a label in `y`. To turn time series classifiers into "global models" - or, equivalently, fine-tunable ones - we would need to be able to pass the entire "local" training batch, `X` and `y`, to `predict` and `predict_proba`. But, there already is an `X` in `predict`...